### PR TITLE
Move iOS 12 CI build to GitHub Actions

### PR DESCRIPTION
**Pull request by NickEntin**
Travis CI [no longer has free macOS builds](https://blog.travis-ci.com/2020-11-02-travis-ci-new-billing), so it's time to migrate away. This moves our iOS 12 snapshot CI build over to GitHub Actions.

### CODE REVIEW

**Comment by NickEntin** [comment]
Had to pin SnapshotTesting to 1.7 in the sample app because 1.8 dropped support for Xcode 10. Really wish that had been considered a breaking change. 😞  Unfortunately GitHub Actions doesn't have an easy way to run iOS 12 tests on Xcode 11, so we need to keep up Xcode 10 support.

**Comment by NickEntin** [comment]
I already marked the Travis CI build as a non-required check since it isn't working properly. Once this is merged I'll mark the new iOS 12 build as required.

- [ ] Add `Xcode Build (iOS_12)` as required PR check

File: Example/Podfile [hunk 1]
```diff
@@ -12,6 +12,10 @@ target 'AccessibilitySnapshotDemo' do
 
 		pod 'AccessibilitySnapshot/iOSSnapshotTestCase', :path => '../AccessibilitySnapshot.podspec'
 		pod 'AccessibilitySnapshot/SnapshotTesting', :path => '../AccessibilitySnapshot.podspec'
+
+		# SnapshotTesting dropped support for building with Xcode 10 in 1.8.0, so pin the version to 1.7 in order to
+	    # run our tests against Xcode 10.
+	    pod 'SnapshotTesting', '= 1.7.2'
```
Comment (NickEntin) [inline Example/Podfile:18]:
```suggestion
		# SnapshotTesting dropped support for building with Xcode 10 in 1.8.0, so pin the version to 1.7 in order to
		# run our tests against Xcode 10.
		pod 'SnapshotTesting', '= 1.7.2'
```

**Review by allisonmoyer — APPROVED** [review]

**Comment by NickEntin** [comment]
We've added the SwiftUI tests on `master` since this PR was opened. The CI build rebases before building, so it was pulling in the SwiftUI changes which don't compile on Xcode 10. I added some Swift version checks to only compile those on Xcode 11+. This shouldn't affect anything since those tests are marked as iOS 13+ anyway.

**Comment by fbernutz** [comment]
Sorry for the issues with this because of the SwiftUI views. I took a look at it and I could get the tests run locally (Xcode 10.3, 11.3.1 and 12.2) with the following changes: 
- Remove the `-weak_framework` for SwiftUI in the "Other Linker Flags" for the Example Project again
- Add another `#if swift(>=5.1)` flag around both SwiftUI Tests
- Update all `#if swift(>=5.1)` flags to `#if swift(>=5.1) && canImport(SwiftUI)`

**Comment by NickEntin** [comment]
Oh awesome, that's a super clean fix. Thanks @fbernutz! No worries about the issues, the required Xcode version support was kind of vague.

**Pull request merged**